bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+
+        It raises following MetaBEXceptions:
+        * DB_DOES_NOT_EXISTS if database dbName does not exist
+        * NOT_MATCHING if the database schema and ascii schema don't match
+        * TB_NOT_IN_DB if the table is described in ascii schema, but it is missing
+                       in the database
+        * COL_NOT_IN_TB if the column is described in ascii schema, but it is
+                        missing in the database
+        * COL_NOT_IN_FL if the column is in the database schema, but not in ascii
+                        schema
+        * Db object can throw various DbException and MySQL exceptions
+        """
+
+        # Connect to the server that has database that is being added
+        db = Db(read_default_file=dbMysqlAuthF)
+        if not db.dbExists(dbName):
+            self._log.error("Db '%s' not found.", dbName)
+            raise MetaBException(MetaBException.DB_DOES_NOT_EXIST, dbName)
+
+        # Parse the ascii schema file
+        x = SchemaToMeta(schemaFile)
+        theTable = x.parse()
+
+        # Fetch the schema information from the database
+        ret = db.execCommandN(
+            "SELECT table_name, column_name, ordinal_position "
+            "FROM information_schema.COLUMNS WHERE "
+            "TABLE_SCHEMA = %s ORDER BY table_name", (dbName,))
+
+        # Count the number of columns in the ascii file
+        nColumns = 0
+
+        for t in theTable:
+            nColumns += len(theTable[t]["columns"])
+
+        # Check if the number of columns matches
+        if nColumns != len(ret):
+            self._log.error("Number of columns in ascii file "
+                       "(%d) != number of columns in db (%d)", nColumns, len(ret))
+            raise MetaBException(MetaBException.NOT_MATCHING)
+
+        # Fetch ordinal_positions from information_schema and add it to "theTable"
+        for (tName, cName, ordP) in ret:
+            t = theTable.get(tName, None)
+            if not t:
+                self._log.error(
+                    "Table '%s' not found in db, present in ascii file.", tName)
+                raise MetaBException(MetaBException.TB_NOT_IN_DB, tName)
+            foundColumn = False
+            for c in t["columns"]:
+                if c["name"] == cName:
+                    foundColumn = True
+                    c["ord_pos"] = int(ordP)
+                    break
+            if not foundColumn:
+                self._log.error(
+                    "Column '%s.%s' not found in db, present in ascii file.",
+                    tName, cName)
+                raise MetaBException(MetaBException.COL_NOT_IN_TB, cName, tName)
+
+        # Check if we covered all columns
+        for t in theTable:
+            for c in theTable[t]["columns"]:
+                if "ord_pos" not in c:
+                    msg = "Column '%s.%s' not found in " % (tName, cName)
+                    msg += "ascii file, present in db."
+                    self._log.error(msg)
```

Comment:
No, definitely I should not be using t here, t is a structure that contains all info about the table, way too much to print.